### ISSUE TITLE
Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by stroke1989** [shiquan/bamdst#22]
Hi, I'm a newer to bamdst.  This is a excellent tool. I runned the example file succefully. However, when I runned the following code:
~/software/bamdst$ /home/ug0416/software/bamdst/bamdst -p /home/ug0416/mydata/genome/hg19/hg19_bed/IDTV2.chr.bed -o /home/ug0416/software/bamdst/coverage/ /home/ug0416/mydata/04.mapping/2nd_geneseed_mapping/SCA_876_T.sorted.bam
error occurred:
Segmentation fault (core dumped)
<img width="920" alt="bamdst" src="https://user-images.githubusercontent.com/61864553/173170539-0b393fdf-5fee-4610-9a38-2c04fdabc1a4.png">
Hope your repose! Appreciate

**Comment by shiquan** [comment]
hi, can you give me a demo to repeat this issue? Thx

**Comment by stroke1989** [comment]
链接：https://pan.baidu.com/s/10LO66IMXjMclHDdYpXM6pw 
提取码：m2rw 
--来自百度网盘超级会员V1的分享
the line include bam file and bed file. Hope your response

**Comment by shiquan** [comment]
I didn't test your data; it's huge for me to download all of it. But I found the chromosome name in the bed is inconsistent with BAM.  You should check that.

**Comment by stroke1989** [comment]
> 
yes, as you point out, I have revised chromosome name in bed file, the attachement is my bed file. However, again., error occurred:
(base) ug0416@gs72:~/software/bamdst$ /home/ug0416/software/bamdst/bamdst -p /home/ug0416/mydata/genome/hg19/hg19_bed/IDTV2.chr.bed -o /home/ug0416/software/bamdst/coverage/ /home/ug0416/mydata/05.gatk/a.BQSR/2nd_BQSR/SCA_271_N.sorted.markdup.bam
Segmentation fault (core dumped)
<img width="204" alt="bedfile" src="https://user-images.githubusercontent.com/61864553/173182776-85b3d96b-590e-4dea-8796-01b7eb577b9a.png">
Hope youre reponse!Appreciate!

**Comment by duolin1986** [comment]
> Hi, I'm a newer to bamdst. This is a excellent tool. I runned the example file succefully. However, when I runned the following code: ~/software/bamdst$ /home/ug0416/software/bamdst/bamdst -p /home/ug0416/mydata/genome/hg19/hg19_bed/IDTV2.chr.bed -o /home/ug0416/software/bamdst/coverage/ /home/ug0416/mydata/04.mapping/2nd_geneseed_mapping/SCA_876_T.sorted.bam error occurred: Segmentation fault (core dumped) <img alt="bamdst" width="920" src="https://user-images.githubusercontent.com/61864553/173170539-0b393fdf-5fee-4610-9a38-2c04fdabc1a4.png"> Hope your repose! Appreciate

same with you

**Comment by shiquan** [comment]
Fixed. Please pull the updated repo.

**Comment by stroke1989** [comment]
> Fixed. Please pull the updated repo.

Great!. Now, it can run successfully

**Comment by ZhangJiePKU** [comment]
Hi, a small question:
Does Bamdst need 1-based bed file？ I wanted to calculate the depth for whole genome using 0-based bed, but I got a same error as stroke1989. 
Thanks.

**Comment by shiquan** [comment]
How many base pairs in your bed file? I didn't test bamdst with a WGS bam, but I think the memory limitation may cause some issue. Maybe to shrink your bed file or split it by chromosomes can help.
